bin/tx-prover/src/proxy/mod.rs

```diff
@@ -40,191 +47,200 @@ impl LoadBalancer {
         }
     }
 
-    /// Create a 429 response for too many requests
-    pub async fn create_too_many_requests_response(
-        session: &mut Session,
-        max_request_per_second: isize,
-    ) -> Result<bool> {
-        // Rate limited, return 429
-        let mut header = ResponseHeader::build(429, None)?;
-        header.insert_header("X-Rate-Limit-Limit", max_request_per_second.to_string())?;
-        header.insert_header("X-Rate-Limit-Remaining", "0")?;
-        header.insert_header("X-Rate-Limit-Reset", "1")?;
-        session.set_keepalive(None);
-        session.write_response_header(Box::new(header), true).await?;
-        Ok(true)
+    /// Get an available worker
+    ///
+    /// This method will return the first available worker from the list of available workers, and
+    /// remove it from the list.
+    /// If no worker is available, it will return None.
+    pub async fn get_available_worker(&self) -> Option<Backend> {
+        self.available_workers.write().await.pop()
     }
 
-    /// Create a 503 response for a full queue
-    pub async fn create_queue_full_response(session: &mut Session) -> Result<bool> {
-        // Set grpc-message header to "Too many requests in the queue"
-        // This is meant to be used by a Tonic interceptor to return a gRPC error
-        let mut header = ResponseHeader::build(503, None)?;
-        header.insert_header("grpc-message", "Too many requests in the queue".to_string())?;
-        header.insert_header("grpc-status", RESOURCE_EXHAUSTED_CODE)?;
-        session.set_keepalive(None);
-        session.write_response_header(Box::new(header.clone()), true).await?;
-
-        let mut error = Error::new(ErrorType::HTTPStatus(503))
-            .more_context("Too many requests in the queue")
-            .into_in();
-        error.set_cause("Too many requests in the queue");
-
-        session.write_response_header(Box::new(header), false).await?;
-        Err(error)
+    /// Set an available worker
+    ///
+    /// This method will add a worker to the list of available workers.
+    /// If the worker is already available, it will panic.
+    pub async fn add_available_worker(&self, worker: Backend) {
```

Comment:
Line 59 needs to be updated. But also we could probably re-write this comment (and other similar related comments) as:
```
/// Adds the provided worker to the list of available workers.
///
/// # Panics
/// Panics if the provided worker is already in the list of available workers.
```